R/FilterState.R

```diff
@@ -399,35 +402,34 @@ FilterState <- R6::R6Class( # nolint
     #'
     ui = function(id) {
       ns <- NS(id)
-      fluidPage(
-        include_css_files(pattern = "filter-panel"),
-        theme = get_teal_bs_theme(),
-        fluidRow(
-          column(
-            width = 10,
-            class = "no-left-right-padding",
+
+      tags$li(
+        id = id,
+        tags$div(
+          class = "filter-card",
+          tags$div(
+            class = "filter-card-header",
             tags$div(
-              tags$span(self$get_varname(),
-                class = "filter_panel_varname"
-              ),
-              if (checkmate::test_character(self$get_varlabel(), min.len = 1) &&
-                tolower(self$get_varname()) != tolower(self$get_varlabel())) {
-                tags$span(self$get_varlabel(), class = "filter_panel_varlabel")
-              }
-            )
+              class = "filter-card-title",
+              tags$span(self$get_varname(deparse = TRUE)),
+              if (length(self$get_varlabel())) {
```

Comment:
The varname and varlabel show twice when it holds the same value:

![image](https://user-images.githubusercontent.com/8597300/217091419-4691a249-e378-46c3-959d-2535a9f34862.png)

In the previous state of the code, we have the if statement comparing `$get_varname` and `$get_varlabel`, which will work, but I don't think this is the root cause.

private$varlabel is initialized with condition to not hold the same value with varname:
https://github.com/insightsengineering/teal.slice/blob/aab1255ec1cbb6442f7d3cef1271eed873aec0a7/R/FilterState.R#L91-L96

However, when I `browser()` this, I see the following:
![image](https://user-images.githubusercontent.com/8597300/217097391-7b1cb4da-5e6f-4e29-960f-92318dc29b1a.png)

varlabel is the result of `get_varlabels()` method from FilteredDataset class:
https://github.com/insightsengineering/teal.slice/blob/aab1255ec1cbb6442f7d3cef1271eed873aec0a7/R/FilteredDatasetDefault.R#L41-L50

I think we need to review this method:
https://github.com/insightsengineering/teal.slice/blob/aab1255ec1cbb6442f7d3cef1271eed873aec0a7/R/FilteredDataset.R#L292-L301

Reply:
Maybe something changes in `formatters::var_labels`?

Reply:
Wow, thanks for the details! I'll look into this.